### ISSUE TITLE
ECS interfaces use native types instead of encapsulating with Output

### ISSUE THREAD

**Issue by tmehlinger** [pulumi/pulumi-awsx#453]
I ran into an issue today trying to pass secrets (which appears to be a [relatively new feature](https://github.com/pulumi/pulumi-awsx/pull/423)) to a container definition using the awsx API. The underlying [`aws.ecs.Container`](https://github.com/pulumi/pulumi-aws/blob/master/sdk/nodejs/ecs/container.ts#L19) interface doesn't encapsulate its properties with `Output`, causing type errors when applying stack configuration.

This is perhaps easiest to illustrate with an abbreviated example:
```typescript
const parameter = new aws.ssm.Parameter("SECRET", {
    type: "SecureString",
    value: config.requireSecret("myPulumiSecret"),
});

const service = new awsx.ecs.FargateService("service", {
    cluster,
    desiredCount: 2,
    taskDefinitionArgs: {
        container: {
            secrets: [{
                name: parameter.name,
                valueFrom: parameter.arn, // FAILS, Type 'Output<string>' is not assignable to type 'string'.
            }],
        },
    }
});
```

I was able to work around the problem with the following (credit to @mikhailshilkov and @CyrusNajmabadi):
```typescript
let deps = [parameter.name, parameter.arn]
const service = pulumi.all(deps).apply(([name, arn]) => {
    return new awsx.ecs.FargateService("service", {
        cluster,
        desiredCount: 2,
        taskDefinitionArgs: {
            container: {
                secrets: [{
                    name: name,
                    valueFrom: arn,
                }],
            },
        }
    });
});
```

**Comment by Adam-Burke** [comment]
Does the above work? It seems to not create the inner resource (e.g. fargate) when I try to do the same with a custom resource. How can I get a string of the arn here?

**Comment by clstokes** [comment]
Related to https://github.com/pulumi/pulumi-aws/pull/1054.

In fact, I think this issues actually belongs in pulumi-aws not awsx?

**Comment by lukemundy-vgw** [comment]
I ran into the same problem, initially I did something similar to you but it's not recommended to create resources in an `apply()` since it means you won't see that resource during previews.

The solution I ended up with is similar though:

```typescript
import * as pulumi from "@pulumi/pulumi";
import * as aws from "@pulumi/aws";
import * as awsx from "@pulumi/awsx";

const config = new pulumi.Config();
const cluster = new awsx.ecs.Cluster("my-cluster");

const ssmSecret = new aws.ssm.Parameter("my-ssm-secret", {
    type: "SecureString",
    value: config.requireSecret("superSecretValue"),
});

const secretsManagerSecret = new aws.secretsmanager.Secret("my-secrets-manager-secret", {
    description: "Top secret!!!"
});

const secretsManagerSecretValue = new aws.secretsmanager.SecretVersion("my-secrets-manager-secret-value", {
    secretId: secretsManagerSecret.id,
    secretString: config.requireSecret("anotherSuperSecretValue"),
}, { parent: secretsManagerSecret });

const service = new awsx.ecs.FargateService("service", {
    cluster,
    desiredCount: 2,
    taskDefinitionArgs: {
        container: {
            image: "nginx:latest",
            secrets: pulumi.all([ssmSecret.arn, secretsManagerSecret.arn])
                .apply(([ssmArn, smArn]) => [
                    { name: "SECRET_VALUE_ONE", valueFrom: ssmArn },
                    { name: "SECRET_VALUE_TWO", valueFrom: smArn },
                    // etc...
                ])
        },
    }
});
```

Since the `secrets` property accepts an `Output<aws.ecs.Secret[]>` you can just generate the array in an `apply()`.

This is an acceptable workaround for me but I definitely agree that the underlying type for `aws.ecs.Secret` should be updated to support an `Output<string>` for the `valueFrom` property.

**Comment by riisi** [comment]
Building on @lukemundy-vgw 's approach, added a helper function to avoid repeating the all/applies.

```
import * as pulumi from "@pulumi/pulumi";
import * as aws from "@pulumi/aws";
import * as awsx from "@pulumi/awsx";

const config = new pulumi.Config();
const cluster = new awsx.ecs.Cluster("my-cluster");

const ssmSecret = new aws.ssm.Parameter("my-ssm-secret", {
    type: "SecureString",
    value: config.requireSecret("superSecretValue"),
});

const secretsManagerSecret = new aws.secretsmanager.Secret("my-secrets-manager-secret", {
    description: "Top secret!!!"
});

const service = new awsx.ecs.FargateService("service", {
    cluster,
    desiredCount: 2,
    taskDefinitionArgs: {
        container: {
            image: "nginx:latest",
            secrets: transformSecrets([
              { name: "SECRET_VALUE_ONE", valueFrom: ssmSecret.arn },
              { name: "SECRET_VALUE_TWO", valueFrom: secretsManagerSecret.arn },
              { name: "SECRET_VALUE_THREE", valueFrom: 'some other string' },
              // etc...
          ])
        },
    }
});

function transformSecrets(
  secrets: Array<{ name: string; valueFrom: pulumi.Output<string> | string }>
) {
  return pulumi
    .all(secrets)
    .apply((transformedSecrets) => transformedSecrets);
}
```

**Comment by danielrbradley** [comment]
I believe this has now been fixed as of the 1.0.0 release. All these types in the SDK are now inputs:

https://github.com/pulumi/pulumi-awsx/blob/bf6f8cdfeaefa2f3f789c676c42c95fd9eebbe6d/sdk/nodejs/types/input.ts#L1382-L1385
